### PR TITLE
TeacherWelcomeScreen + Modal Image Selection Grid enhancements

### PR DESCRIPTION
**Pull request by ElyDev**
- Implemented Teacher Welcome Screen

- A number of components:

1.  TouchableText for text style button
2. - TouchableAvatar: for an avatar circular image that responds to press events (wrapped in TouchableHighlight)
3. - ImageSelectionGrid to render avatars over a grid and select pressed image
4. - ImageSelectonModal: Modal dialog that wraps ImageSelectionGrid + a cancel button
5. - ImageSelectionRow: A row of few suggested images + a button to expand to the Modal grid and select from more selections. The row also causes selected images to expand

- Updated TeacherProfileScreen and AddClassScreen to use the common components above
- Updated those screens to use KeyboardAvoidingView to adjust view to keyboard invocation
- Updated snapshot tests and added snapshot test for teacher welcome and profile pages

- 
Screenshot_1553407876

### CODE REVIEW

File: Dev/QConnect/src/screens/TeacherScreens/TeacherProfile/TeacherProfileScreen.js [hunk 1]
```diff
@@ -36,59 +48,53 @@ export class TeacherProfileScreen extends Component {
...
-                        onChangeText={newText =>
-                            this.setState({
-                                name: this.state.name,
-                                phoneNumber: this.state.phoneNumber,
-                                emailAddress: newText
-                        })} 
-                        value={this.state.emailAddress}/>
-                    </View>
-                </View>
+
```
Comment (zyadelgohary) [inline Dev/QConnect/src/screens/TeacherScreens/TeacherProfile/TeacherProfileScreen.js]: I'm confused why this change, because in all of our classes we have the UI rendering as part of the screen.js file. Why put it all in a separate "TeacherInfoEntries" file.

Comment (ElyDev) [reply Dev/QConnect/src/screens/TeacherScreens/TeacherProfile/TeacherProfileScreen.js]: This is a separate component now because it a shared UI between the Teacher Welcome screen and the Teacher Profile Screen. They both render the text input fields related to teacher profile info. So I wanted to be consistent in our UI as well as reduce code duplication.

Comment (zyadelgohary) [reply Dev/QConnect/src/screens/TeacherScreens/TeacherProfile/TeacherProfileScreen.js]: Got it, thanks for the clarification.
